test/units/cli/test_vault.py

```diff
@@ -142,6 +141,20 @@ def test_encrypt_string_stdin(self, mock_stdin_read, mock_vault_editor, mock_set
         cli.parse()
         cli.run()
 
+    @patch('ansible.cli.vault.VaultCLI.setup_vault_secrets')
+    @patch('ansible.cli.vault.VaultEditor')
+    def test_encrypt_string_stdin_prompt_error(self, mock_vault_editor, mock_setup_vault_secrets):
+        mock_setup_vault_secrets.return_value = [('default', TextVaultSecret('password'))]
+        cli = VaultCLI(args=['ansible-vault',
+                             'encrypt_string',
+                             '--stdin-name',
+                             'the_var_from_stdin',
+                             '-',
+                             '--prompt'])
+        self.assertRaisesRegexp(errors.AnsibleOptionsError,
+                                "The --prompt option is not supported if also reading input from stdin",
+                                cli.parse)
```

Comment:
We use `pytest` to run tests and the helpers from `unittest.TestCase` should be avoided (ideally, `unittest.TestCase` should never be used because it limits the compatibility with `pytest`). Instead, use `pytest.raises()`:
```suggestion
        expected_error_msg = (
            r'^The --prompt option is not supported if '
            r'also reading input from stdin$'
        )
        with pytest.raises(errors.AnsibleOptionsError, match=expected_error_msg):
            cli.parse()
```